core/src/main/java/io/questdb/cairo/mv/MatViewRefreshExecutionContext.java

```diff
@@ -82,16 +87,28 @@ public TableReader getReader(TableToken tableToken, long version) {
                         baseTableReader.getMetadataVersion()
                 );
             }
-            return baseTableReader;
+            // Fast path: go with the base reader if it's not in-use.
+            if (getCairoEngine().getDetachedReaderRefCount(baseTableReader) == 0) {
```

Comment:
nit: duplicate code can be hidden inside reader pool without exposing these all of these methods?

Reply:
Done in 6385339fd1b1514f70520e5c8f230ef8f37eec39